packages/graphql-codegen-cli/src/utils/watcher.ts

```diff
@@ -142,7 +142,7 @@ export const createWatcher = (
 
   // the promise never resolves to keep process running
   return new Promise<void>((resolve, reject) => {
-    executeCodegen(initalContext)
+    (config.watchConfig?.skipFirst ? Promise.resolve([]) : executeCodegen(initalContext))
```

Comment:
It might be incorrect to return an array in here, but since by default initial context resulted with empty array - it was pretty clear to me that I should use the same, although i'm not happy with it. Maybe we can unify it with (let's say) some default via function or some other way around?